### PR TITLE
[Distributed]fix code smell

### PR DESCRIPTION
**Pull request by neuyilan**

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/log/LogDispatcher.java [hunk 1]
```diff
@@ -244,25 +244,20 @@ private void appendEntriesAsync(List<ByteBuffer> logList, AppendEntriesRequest r
...
-        start = System.nanoTime();
-      }
+      Timer.Statistic.RAFT_SENDER_WAIT_FOR_PREV_LOG.setStartTime();
       if (!member.waitForPrevLog(peer, currBatch.get(0).getLog())) {
         logger.warn("{}: node {} timed out when appending {}", member.getName(), receiver,
             currBatch.get(0).getLog());
         return;
       }
-      Timer.Statistic.RAFT_SENDER_WAIT_FOR_PREV_LOG.addNanoFromStart(start);
+      Timer.Statistic.RAFT_SENDER_WAIT_FOR_PREV_LOG.calCostTime();
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/log/LogDispatcher.java:253]: I don't think the change is thread-safe, please confirm it.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/Timer.java [hunk 2]
```diff
@@ -183,14 +184,33 @@ public void add(long val) {
...
-    public void addNanoFromStart(long start) {
+    public void setStartTime() {
+      setStartTime(System.nanoTime());
+    }
+
+    public void setStartTime(long startTime) {
+      if (ENABLE_INSTRUMENTING) {
+        this.startTime = startTime;
+      }
+    }
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/Timer.java:197]: The condition should be put into the first method, as `System.nanoTime()` is a heavy call and should be avoided when not using instrumenting.

**Review by jt2594838 — CHANGES_REQUESTED** [review]

File: cluster/src/main/java/org/apache/iotdb/cluster/log/LogDispatcher.java (hunk 1 above)
Comment (neuyilan) [reply cluster/src/main/java/org/apache/iotdb/cluster/log/LogDispatcher.java:253]: you're right, its indeed cause concurrent problem, I'll fix it. thanks!

File: cluster/src/main/java/org/apache/iotdb/cluster/server/Timer.java (hunk 2 above)
Comment (neuyilan) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/Timer.java:197]: Got it

**Review by jt2594838 — APPROVED** [review]

**Pull request merged**